### PR TITLE
LikeCoin Button 2.0

### PR DESCRIPTION
**Pull request by nwingt**

### CODE REVIEW

File: src/components/LikeButtonV2/LikeButtonV2.vue [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+    return {
+      radius: 38,
+      hasBlockingAnimation: false,
+      clapBits: [],
+      isShowStarBits: false,
+    };
+  },
+  computed: {
+    state() {
+      if (this.count < 5) {
```
Comment (williamchong007) [inline src/components/LikeButtonV2/LikeButtonV2.vue:263]: Should be constant

File: src/components/LikeButtonV2/LikeButtonV2.vue [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+      }
+      return 'superlikeable';
+    },
+    isDisabled() {
+      return this.state === 'cooldown'
+        || this.state === 'superlikeable-cooldown'
+        || this.hasBlockingAnimation;
+    },
+    isShowBadge() {
+      return this.count >= 1;
```
Comment (williamchong007) [inline src/components/LikeButtonV2/LikeButtonV2.vue:286]: prefer `> 0`

File: src/components/LikeButtonV2/LikeButtonV2.vue [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+        case 'superliked':
+          return '#28646e';
+
+        case 'cooldown':
+        case 'superlikeable-cooldown':
+          return '#50e3c2';
+
+        case 'initial':
+        default:
+          if (this.isPressing || this.count >= 1) {
```
Comment (williamchong007) [inline src/components/LikeButtonV2/LikeButtonV2.vue:334]: prefer `> 0`, also if this keep coming up, maybe should use computed

**Comment by williamchong007** [comment]
This PR is only UI no API right?

**Comment by nwingt** [comment]
> This PR is only UI no API right?

@williamchong007 Yes

File: src/components/LikeButtonV2/LikeButtonV2.badge.vue [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    viewBox="0 0 36 36"
+    width="36"
+  >
+    <transition
+      @before-enter="beforeEnter"
+      @enter="enter"
+      @leave="leave"
+      :css="false"
+    >
+      <g v-if="!isHidden">
```
[the remark this answers is not in the document]
Comment (nwingt) [reply src/components/LikeButtonV2/LikeButtonV2.badge.vue:15]: Not sure what u mean, but this boolean is for toggling animation

File: src/components/LikeButtonV2/LikeButtonV2.clapBits.vue [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      required: true,
+    },
+    explosionRange: {
+      type: Number,
+      required: true,
+    },
+    explosionSize: {
+      type: Number,
+      required: true,
+    },
```
[the remark this answers is not in the document]
Comment (nwingt) [reply src/components/LikeButtonV2/LikeButtonV2.clapBits.vue:44]: These two props are only for Storybook.js use, it very much depends on the parent component who use it, so I set required to alert people to use it wisely

**Pull request merged**